Visitor remodel

andrewobler opened the pull request:
Remodel for main_worker to use a visitor model instead of a messy switch statement. Also adds
some different handling/config infrastructure for Titan, rather than just Luna.

Still TODO: implement a similar visitor for main_network_worker.

andrewobler committed:
Starting point, added some basic infrastructure

andrewobler committed:
Implemented visitor for main_worker

andrewobler committed:
Added support for Titan valve configurations

andrewobler committed:
Set TANK valve closed to HIGH

andrewobler committed:
Fixed VENT open and close

ty19 committed:
Add pre-sets to the processes and types.

andrewobler committed:
Moved some files around, visitor imps now in own file

ty19 committed:
Add Titan default preset

andrewobler committed:
Touch ups, added new visitProc for Titan visitor

ty19 commented:
The fact that queue_visitor_imps contains both the main/luna and titan visitors is kind of confusing. It would be better if the two visitors were separated into their own files. It also makes more sense for titan_work_queue_visitor to inherit from work_queue_visitor instead of main_work_queue_visitor

Nitpicky changest: change main_visitor, etc. to luna_visitor. Move the visitors to their own directory and not server/.

Otherwise, looks good, aside from some unfortunate code duplication with the processes and visit_timed. That's a fix for another time though.

ty19 committed:
Move titan and luna into individual files.

Now in the src/server/ directory

ty19 committed:
Fix to make things compile.

Moved adc_data, config vars from main_worker into worker_visitor.
Moved visitNone, visitDefault from luna/titan_visitor to worker_visitor.

ty19 committed:
Move visitProc, visitTimed, visitIgn into parent

visitProc, visitTimed in luna/titan now have default cases that
call the parent implementation for shared processes/timed items.

Current issues:

The Titan visitor ign2 had MAIN open. My info says MAIN should be
closed during ign2. Maybe I just copied the wrong things when
moving the files around.

worker_visitor.hpp shouldn't have gitvc vars since those are luna
specific. Maybe we should move this into ini_config.hpp, but it's
not necessary for correctness.

preignite_us, hotflow_us, gitvc_wait_time are hackily set in
worker_visitor.hpp to prevent a circular include with
timed_item_list.hpp. They're also int instead of timestamp_t.

ty19 commented:
cf9dbbe commit message is wrong. Was in src/server/, now in src/visitor/

Overall, moved titan/luna files into a new directory and reduced code duplication by moving things into the parent class, worker_visitor.hpp. Builds and mocked runs but have not tested with actual setup.

See 8ea7a18 commit message for details.

ty19 committed:
Fix compilation on Pi

Also added titan_oedk_hotflow.ini which was used for
basic tests at the oedk.

ty19 reviewed (approved):
looks good
